File: python/2016/let-me-make-you-a-martyr-add.py

from movies import MovieRepository

# Create a new movie repository using the default configuration
movies = MovieRepository()

# Add Let Me Make You a Martyr to the database
movies.insert(
    title="Let Me Make You a Martyr", 
    year=2016, 
    plot="A cerebral revenge film about two adopted siblings who fall in love, and hatch a plan to kill their abusive father.", 
    rating=6.4
)

# Confirm that the movie was added
movie = movies.select(
    title="Let Me Make You a Martyr", 
    year=2016
)
if movie:
    # The movie was found
    print(f"Movie found: {movie}")
else:
    # The movie was not found
    print("Movie not found")
    

File: csharp/2016/let-me-make-you-a-martyr-add.cs
using System;
using System.Threading.Tasks;
using Amazon.DynamoDBv2;
using Amazon.DynamoDBv2.Model;
using System.Collections.Generic;

namespace MovieDatabase
{
    class Program
    {
        static async Task Main(string[] args)
        {
            // Create a MovieRepository instance
            var movies = new MovieRepository();

            // Add "Let Me Make You a Martyr" to the database
            // This demonstrates how to insert a new item into DynamoDB
            await movies.InsertAsync(
                title: "Let Me Make You a Martyr",
                year: 2016,
                plot: "A cerebral revenge film about two adopted siblings who fall in love, and hatch a plan to kill their abusive father.",
                rating: 6.4
            );

            // Confirm that the movie was added by retrieving it
            var movie = await movies.SelectAsync(
                title: "Let Me Make You a Martyr",
                year: 2016
            );

            if (movie != null)
            {
                // The movie was found
                Console.WriteLine($"Movie found: {movie}");
            }
            else
            {
                // The movie was not found
                Console.WriteLine("Movie not found");
            }
        }
    }
}
